Cargo.toml

```diff
@@ -27,10 +27,17 @@ bytes = "1"
 structopt = "0.3.14"
 tokio = { version = "1", features = ["full"] }
 tokio-stream = "0.1"
-tracing = "0.1.13"
+tracing = "0.1.34"
 tracing-futures = { version = "0.2.3" }
-tracing-subscriber = "0.2.2"
+tracing-subscriber = { version = "0.3.11", features = ["env-filter"] }
```

Comment:
Since this repo is for learning, it would be super useful to comment each of these dependencies with a quick bit on why they are needed.

Reply:
sounds good